File: wehrdj/connect.py
import datajoint as dj
import json

def connect(host=None, user=None, password=None, file=None):
	if file is not None:
		with open(file, 'r') as cfgfile:
			cfg = json.load(cfgfile)
		host = cfg['host']
		user = cfg['user']
		password = cfg['password']
	else:
		if host is None:
			host = input('host ip and port')
		if user is None:
			user = input('username')
		if password is None:
			password = input('password')

	dj.config['database.host'] = host
	dj.config['database.user'] = user
	dj.config['database.password'] = password

File: matlab/connect.m
function conn = connect()
%connect to wehrlab datajoint database

fname = '~/.djcredentials.json'; 
fid = fopen(fname); 
raw = fread(fid,inf); 
str = char(raw'); 
fclose(fid); 
djcred = jsondecode(str);
setenv('DJ_HOST', djcred.host)
setenv('DJ_USER', djcred.user)
setenv('DJ_PASS', djcred.password)

%dj.config('databasePort', 3307)
dj.config('databaseUse_tls', false)
dj.config
conn=dj.conn()